persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideImpl.scala

```diff
@@ -62,65 +62,52 @@ private[lagom] class ReadSideImpl @Inject()(
       clazz: Class[_]
   ) = {
 
-    // Only run if we're configured to run on this role
-    if (config.role.forall(Cluster(system).selfRoles.contains)) {
-      // try to create one instance to fail fast (e.g. wrong constructor)
-      val dummyProcessor = try {
-        processorFactory()
-      } catch {
-        case NonFatal(e) =>
-          throw new IllegalArgumentException(
-            "Cannot create instance of " +
-              s"[${clazz.getName}]",
-            e
-          )
-      }
-
-      val readSideName           = name.asScala.fold("")(_ + "-") + dummyProcessor.readSideName()
-      val encodedReadSideName    = URLEncoder.encode(readSideName, "utf-8")
-      val tags                   = dummyProcessor.aggregateTags().asScala
-      val entityIds: Set[String] = tags.map(_.tag).toSet
-      val eventClass = tags.headOption match {
-        case Some(tag) => tag.eventType
-        case None      => throw new IllegalArgumentException(s"ReadSideProcessor ${clazz.getName} returned 0 tags")
-      }
-
-      val globalPrepareTask: ClusterStartupTask =
-        ClusterStartupTask(
-          system,
-          s"readSideGlobalPrepare-$encodedReadSideName",
-          () => processorFactory().buildHandler().globalPrepare().toScala,
-          config.globalPrepareTimeout,
-          config.role,
-          config.minBackoff,
-          config.maxBackoff,
-          config.randomBackoffFactor
-        )
-
-      val streamName     = tags.head.eventType.getName
-      val projectionName = readSideName
-
-      val readSidePropsFactory = (projectionRegistryActorRef: ActorRef) =>
-        ReadSideActor.props(
-          streamName,
-          projectionName,
-          config,
-          eventClass,
-          globalPrepareTask,
-          persistentEntityRegistry.eventStream[Event],
-          processorFactory,
-          projectionRegistryActorRef
-        )
-
-      projectionRegistryImpl.registerProjectionGroup(
-        tags.head.eventType.getName, // TODO: use the name from the entity, not the tags
-        entityIds,
-        readSideName,
+    val readSideProcessor      = processorFactory()
```

Comment:
On line 92 (unchanged) we are calling the `processorFactory()` again. We could re-use `readSideProcessor`.

Reply:
Are you referring to this code: (line numbers got scrambled)

```scala
    val globalPrepareTask: ClusterStartupTask =
      ClusterStartupTask(
        system,
        s"readSideGlobalPrepare-$encodedReadSideName",
        () => processorFactory().buildHandler().globalPrepare().toScala,
        config.globalPrepareTimeout,
        config.role,
        config.minBackoff,
        config.maxBackoff,
        config.randomBackoffFactor
      )
```

Reusing the `readSideProcessor` instance would be closing on it. 🤔
I guess there's nothing wrong about it. I'll change it, yeah.